src/flake8_pytest_style/plugins/parametrize.rs

```diff
@@ -32,11 +34,34 @@ fn check_names(checker: &mut Checker, expr: &Expr) {
                             Range::from_located(expr),
                         );
                         if checker.patch(check.kind.code()) {
-                            check.amend(Fix::replacement(
-                                strings_to_python_tuple(&names),
-                                expr.location,
-                                expr.end_location.unwrap(),
-                            ));
+                            let mut generator = SourceCodeGenerator::new(
+                                checker.style.indentation(),
+                                checker.style.quote(),
+                                checker.style.line_ending(),
+                            );
+                            generator.unparse_expr(
+                                &create_expr(ExprKind::Tuple {
+                                    elts: names
+                                        .iter()
+                                        .map(|name| {
+                                            create_expr(ExprKind::Constant {
+                                                value: Constant::Str(name.to_string()),
+                                                kind: None,
+                                            })
+                                        })
+                                        .collect(),
+                                    ctx: ExprContext::Load,
+                                }),
+                                1,
+                            );
+
+                            if let Ok(content) = generator.generate() {
+                                check.amend(Fix::replacement(
```

Comment:
Can you tweak these to log the error on failure? If you grep for similar generate calls, you should be able to find the preferred pattern.

Reply:
Done! Thanks for the quick review. Really enjoying learning Rust. :smile:

Reply:
Thank you!